dbt/adapters/databricks/connections.py

```diff
@@ -410,6 +412,37 @@ def validate_invocation_env(cls, invocation_env: str) -> None:
                 f"Invalid invocation environment: {invocation_env}"
             )
 
+    @classmethod
+    def get_all_http_headers(
+        cls,
+        connection_parameters: Optional[Dict[str, Any]],
+        http_session_headers_str: Optional[str],
+    ) -> List[Tuple[str, str]]:
+        user_http_session_headers: Dict[str, str] = (
+            connection_parameters.pop("http_headers", {})
```

Comment:
Please avoid a side effect on the parameter.
I guess we can just pass `user_http_session_headers: Dict[str, str]` to this method.